### ISSUE TITLE
Activity node for reaction should be a process if GO BP is curated at Reactome

### ISSUE THREAD

**Issue by dustine32** [geneontology/pathways2GO#318]
Here is how the reaction [R-HSA-163743](https://reactome.org/content/detail/R-HSA-163743) "Transcriptional activation of ACACA by ChREBP:MLX" (the middle box) is currently converted:
![image](https://github.com/geneontology/pathways2GO/assets/2678599/9c9b4cfa-af2a-4b6f-a414-666110197ade)

There is no GO MF term or controller so the activity is Molecular Event. Reactome has curated a GO BP term so it is added and connected to the reaction via a "part of" relation.

The new conversion requirements should instead result in this pattern:
![image](https://github.com/geneontology/pathways2GO/assets/2678599/4ee54774-958d-4117-9839-e3c8dad6eb7e)
The Molecular Event activity is gone with the Reactome-curated GO BP term now anchoring the "has input" and "has output" edges and connected to the preceding reaction via a "part of" relation.

In other words:
1. Remove edges from the reaction to its preceding reaction (usually a causal relation)
2. Replace this edge with "part of"
3. Replace reaction term with curated GO BP term
4. Don't create a new individual node for the GO BP

![image](https://github.com/geneontology/pathways2GO/assets/2678599/fc41c49b-4b62-4828-ac0f-4d9d5d6fa9e7)
This rule should apply when a reaction class is Molecular Event (no GO MF term) and has a curated GO BP term.

The code should decide which relation to connect between the reaction's curated BP node and its preceding activity node by fetching the preceding node's MF term and checking if the BP term is in the MF term's "part of" closure (ancestor terms). If yes, the relation will be "part of," otherwise the relation will be "causally upstream of".

**Comment by dustine32** [comment]
@ukemi @deustp01 Plugging in this change and running the tests, our transport test example reaction ([R-HSA-201669](https://reactome.org/content/detail/R-HSA-201669) in model [R-HSA-201681](http://noctua-dev.berkeleybop.org/editor/graph/gomodel:R-HSA-201681?)) has changed because it's curated to a GO BP term. Showing the reaction before my change:
![image](https://github.com/geneontology/pathways2GO/assets/2678599/fd28b71d-5e35-48cf-ac89-cb37cf90d62e)
And after:
![image](https://github.com/geneontology/pathways2GO/assets/2678599/e52f1237-f86f-41c8-be3b-19fea5cd21c4)
As you can see, the location info has been discarded along with the inferred "transport activity" function term. Is this ok with you?

**Comment by ukemi** [comment]
Hi @dustine32. Yes and no. This is something we should discuss on a weeds call. For nuclear import, we have been inferring a transporter activity and hand waving that the enabler would be the nuclear pore. We need to discuss whether this is ok. If not, we have a biological process called 'import into the nucleus'. I think that process is a better asserted match for the black box reaction below that the regulation BP. I will put it on the agenda for a future call.

**Comment by deustp01** [comment]
> I think that process is a better asserted match for the black box reaction below that the regulation BP.

Questions, perhaps to set up that weeds discussion

- The textbook description of getting. cargo across the nuclear envelope via the nuclear pore (in either direction) distinguishes multiple steps enabled by multiple soluble accessory proteins and multiple proteins anchored in the nuclear pore, which sounds like a process, but at the same time under normal physiological conditions once a cargo starts the trip it always completes it, so all these steps can be viewed in the same way as the multiple intermediates formed in the course of hydrolysis of a peptide bond by a serine protease. So, we need a convention. Would the process / blackBoxEvent (analogous to the "gene expression" process convention we are working our way towards for transcriptional regulation of gene expression) work here as well?
- Inferring a transporter activity: again, the textbook has it that everything over a certain size must interact with pore components as above in order to traverse the pore, so it seems like a very conservative inference to assert that protein or RNA above that minimum size follows the textbook route even though it has not been individually studied in experiments. Or am I missing something?

**Comment by ukemi** [comment]
You're not missing anything. I would argue for the import process assertion. I think we are seeing over and over that the inferred transporter activities are often problematic. It was a nice idea, but there are lots of exceptions where what is represented is actually a process. I still think we should discuss it with the other curators.
